src/main/java/seedu/address/logic/commands/EditReviewCommand.java

```diff
@@ -0,0 +1,185 @@
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditReviewDescriptor editReviewDescriptor;
+
+    /**
+     * @param index of the review in the review list to edit
+     * @param editReviewDescriptor details to edit the review with
+     */
+    public EditReviewCommand(Index index, EditReviewDescriptor editReviewDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editReviewDescriptor);
+
+        this.index = index;
+        this.editReviewDescriptor = new EditReviewDescriptor(editReviewDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+
+        requireNonNull(model);
+        List<Review> lastShownList = model.getActiveReviews();
+        Eatery activeEatery = model.getActiveEatery();
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EATERY_DISPLAYED_INDEX);
+        }
+
+        Review reviewToEdit = lastShownList.get(index.getZeroBased());
+        Review editedReview = createEditedReview(reviewToEdit, editReviewDescriptor);
+
+        lastShownList.set(index.getZeroBased(), editedReview);
+        activeEatery.setReviews(lastShownList);
+        return new CommandResult(String.format(MESSAGE_EDITED_REVIEW_SUCCESS), editedReview);
+    }
+
+    /**
+     * Creates and returns a {@code Review} with the details of {@code reviewToEdit}
+     * edited with {@code editReviewDescriptor}.
+     */
+    private static Review createEditedReview(Review reviewToEdit, EditReviewDescriptor editReviewDescriptor) {
+        assert reviewToEdit != null;
+
+        String updatedDescription = editReviewDescriptor.getDescription().orElse(reviewToEdit.getDescription());
+        double updatedCost = editReviewDescriptor.getCost().orElse(reviewToEdit.getCost());
+        int updatedRating = editReviewDescriptor.getRating().orElse(reviewToEdit.getRating());
+        Date updatedDate = editReviewDescriptor.getDate().orElse(reviewToEdit.getDate());
+
+        return new Review(updatedDescription, updatedCost, updatedRating, updatedDate);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof EditReviewCommand)) {
+            return false;
+        }
+
+        EditReviewCommand e = (EditReviewCommand) other;
+        return index.equals(e.index)
+                && editReviewDescriptor.equals(e.editReviewDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the review with. Each non-empty field value will replace the
+     * corresponding field value of the review.
+     */
+    public static class EditReviewDescriptor {
+        private String description;
+        private double cost;
+        private int rating;
```

Comment:
I think you should be using OptionalDouble and OptionalInt instead of just double and int as iirc, double and int are initialised to 0 even if you didn't assign a number to it.

Reply:
Yup this is true, just verified: 

```
jshell> double d;
d ==> 0.0

jshell> int i;
i ==> 0
```